ckan_datapackage_tools/converter.py

```diff
@@ -177,82 +174,169 @@ def _parse_notes(dataset_dict):
     return result
 
 
-def _parse_license(dataset_dict):
+def _parse_tags(dataset_dict):
     result = {}
-    license = {}
 
-    if dataset_dict.get('license_id'):
-        license['type'] = dataset_dict['license_id']
-    if dataset_dict.get('license_title'):
-        license['title'] = dataset_dict['license_title']
-    if dataset_dict.get('license_url'):
-        license['url'] = dataset_dict['license_url']
+    keywords = [tag['name'] for tag in dataset_dict.get('tags', [])]
 
-    if license:
-        result['license'] = license
+    if keywords:
+        result['keywords'] = keywords
 
     return result
 
 
-def _parse_author_and_source(dataset_dict):
+def _parse_extras(dataset_dict):
     result = {}
-    source = {}
-
-    if dataset_dict.get('author'):
-        source['name'] = dataset_dict['author']
-    if dataset_dict.get('author_email'):
-        source['email'] = dataset_dict['author_email']
-    if dataset_dict.get('url'):
-        source['web'] = dataset_dict['url']
+    extras = [[extra['key'], extra['value']] for extra
+              in dataset_dict.get('extras', [])]
+    for extra in extras:
+        try:
+            extra[1] = json.loads(extra[1])
+        except (ValueError, TypeError):
+            pass
+    sources = _parse_organization_as_primary_source(dataset_dict)
+    licenses = _parse_primary_license(dataset_dict)
+    contributors = _parse_primary_contributors(dataset_dict)
+    if extras:
+        extras_dict = dict(extras)
+        result.update(_parse_profile(extras_dict))
+        _parse_extra_sources(extras_dict, sources)
+        _parse_extra_licenses(extras_dict, licenses)
+        _parse_extra_contributors(extras_dict, contributors)
+        if extras_dict:
+            result['extras'] = extras_dict
+    if sources:
+        result['sources'] = sources
+    if licenses:
+        result['licenses'] = licenses
+    if contributors:
+        result['contributors'] = contributors
+    return result
 
-    if source:
-        result['sources'] = [source]
 
+def _parse_profile(dataset_dict):
+    result = {}
+    profile = dataset_dict.pop('profile', '')
+    if profile:
+        result['profile'] = profile
     return result
 
 
-def _parse_maintainer(dataset_dict):
-    result = {}
-    author = {}
+def _parse_organization_as_primary_source(dataset_dict):
+    all_sources = []
+    source = {}
+    organization = dataset_dict.get('organization')
+    if organization.get('is_organization', False):
```

Comment:
`is_organization` should always be True for `dataset_dict['organization']` objects

Reply:
Yep OK, thanks. wasn't sure about this as I saw the reference thought perhaps there might be cases where this is false.